internal/clientconn/otel_propagation.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clientconn
+
+import (
+	"encoding/json"
+
+	"go.opentelemetry.io/otel/trace"
+
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+)
+
+// spanContextFromComment extracts OpenTelemetry tracing information from comment.
+//
+// If the comment is empty, it returns an empty span context and no error.
+func spanContextFromComment(comment string) (trace.SpanContext, error) {
+	if comment == "" {
+		return trace.SpanContext{}, nil
+	}
+
+	type TraceData struct {
+		TraceID [16]byte `json:"ferretTraceID"`
+		SpanID  [8]byte  `json:"ferretSpanID"`
+	}
+
+	var data TraceData
+
+	err := json.Unmarshal([]byte(comment), &data)
+	if err != nil {
+		return trace.SpanContext{}, lazyerrors.Error(err)
```

Comment:
I suppose we don't stop users from setting other comment of their choice. Is this error message sufficient to tell if that happens such as any string.

Reply:
I think it's enough as a prototype :)

Ideally, we should also validate if `TraceID` and `SpanID` are correct. I suggest, we decide first if we do anything with this approach, and then we can have better errors etc.